python/plugins/processing/tools/general.py

```diff
@@ -75,3 +76,26 @@ def runalg(algOrName, *args, **kwargs):
 
 def runandload(name, *args, **kwargs):
     return Processing.runAlgorithm(name, handleAlgorithmResults, *args, **kwargs)
+
+
+def version():
+    """Returns the Processing version number.
+
+    :returns: The version number.
+    :rtype: int
+    """
+    # Get location of application wide version info
+    root_dir = abspath(join(dirname(__file__), '..'))
+    fid = open(join(root_dir, 'metadata.txt'))
```

Comment:
In general it's preferable to use

with open(filename) as file:
    filed.readlines() # or something else

it's shorter and handles exceptions etc.